packages/backend/src/index.ts

```diff
@@ -14,18 +14,112 @@ const TODOS = [
     desc: 'Brush your teeth'
   }
 ]
-
+function generateUniqueId() {
+  return Date.now() + Math.floor(Math.random() * 1000)
+}
 const app = new Elysia()
   .get('/todos', () => TODOS)
   .get(
     '/todos/:id',
     ({ params, error }) => {
       const todo = TODOS.find((todo) => todo.id === params.id)
       if (!todo) {
-        return error(404)
+        return error(404, 'Todo with id ' + params.id + ' not found') // custom error message
+      }
+      return todo
+    },
+    {
+      params: t.Object({
+        id: t.Numeric()
+      })
+    }
+  )
+  .post(
+    '/todos',
+    ({ body, set }) => {
+      const newTodo = {
+        id: generateUniqueId(),
+        ...body
+      }
+      set.status = 201
+      TODOS.push(newTodo)
+      return newTodo
+    },
+    {
+      body: t.Object({
+        desc: t.String({
+          minLength: 1,
+          error: 'Description is required'
+        }),
+        starred: t.Boolean(),
+        completed: t.Boolean()
+      })
+    }
+  )
+  .put(
+    '/todos/:id',
+    ({ params, body, error }) => {
+      const todo = TODOS.find((todo) => todo.id === params.id)
+      if (!todo) {
+        return error(404, `Failed to find todo with id ${params.id} `) // custom error message
       }
+      Object.assign(todo, body)
       return todo
     },
+    {
+      params: t.Object({
+        id: t.Numeric()
+      }),
+      body: t.Object({
+        desc: t.String({
+          minLength: 1,
+          error: 'Description is required'
+        }),
+        starred: t.Boolean(),
+        completed: t.Boolean()
+      })
+    }
+  )
+  .patch(
+    '/todos/:id',
+    ({ params, body, error }) => {
+      const todoToUpdate = TODOS.find((todo) => todo.id === params.id)
+      if (!todoToUpdate) {
+        return error(404, `Failed to find todo with id ${params.id} `) // custom error message
+      }
+      const index = TODOS.indexOf(todoToUpdate)
+      Object.assign(todoToUpdate, body)
+      TODOS[index] = todoToUpdate
```

Comment:
We aren't doing this in case of `PUT` operation so either this is extraneous or the `PUT` endpoint might not be working as expected